omnigibson/objects/object_base.py

```diff
@@ -207,7 +207,7 @@ def articulation_root_path(self):
         if self.kinematic_only or ((not has_articulated_joints) and (not has_fixed_joints)):
             # Kinematic only, or non-jointed single body objects
             return None
-        elif not self.fixed_base:
+        elif not self.fixed_base and has_articulated_joints:
```

Comment:
hmmm I thought we debug this extensively before, is it still buggy? 

What behavior did this code change?

not kinematic_only
not fixed_base
not has_articulated_joints
yes has_fixed_joints

e.g. a cup with a fillable volume meta link.

In this case, what should be the articulation_root_path?

Reply:
The interesting thing is that the comment in the immediately proceeding line suggests that the (fixed) code is actually how the logic _should_ be.

I think originally, non-articulated, non-fixed objects that had metalinks (fixed joints) were treated as kinematic-only objects, until I found out that that caused some USD issues. So I updated the logic so that this category of objects is not satisfied by that first `if` statement, but didn't change anything else -- so I think it, by default, went to the `else` clause. But really it belonged to the `elif` clause, since moving the articulation root path from the object prim to the root link prim is only done out of _necessity_ for articulated, non-fixed objects since their joint ordering gets messed up by omniverse otherwise.

To answer your specific questions, yes, that is the specific case I'm talking about, and the articulation_root_path should remain unchanged (i.e.: exist at the object prim level, NOT the root link prim level)